crates/query-engine/translation/src/translation/mutation/experimental/insert.rs

```diff
@@ -90,14 +87,14 @@ fn translate_object_into_columns_and_values(
             }
             Ok(())
         }
-        serde_json::Value::Array(_) => Err(Error::UnexpectedStructure(
-            "array of arrays structure in insert _objects argument. Expecting an array of objects."
-                .to_string(),
-        )),
-        _ => Err(Error::UnexpectedStructure(
-            "array of values structure in insert _objects argument. Expecting an array of objects."
-                .to_string(),
-        )),
+        serde_json::Value::Array(_) => Err(Error::UnexpectedStructure(format!(
```

Comment:
Why isn't this in the `Display` impl of `UnexpectedStructure`?

Reply:
Your comment made sense to me. I went to have a look what would I need to parameterize UnexpectedStructure by to make this work, and the result was:

- array of arrays
- insert
- {} argument
- an array of objects

Which is essentially almost the entire message :( Might not be worth trying to parameterize in that case, and we probably don't need a dedicated error time for each unexpected structure.